src/System.Security.Cryptography.Algorithms/tests/HashAlgorithmTest.cs

```diff
@@ -54,6 +55,154 @@ private void VerifyICryptoTransformStream(Stream input, string output)
+                actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                Assert.Equal(expectedEmpty, actual);
+
+                // Re-hash the same data
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                actual = hash.ComputeHash(block2, 0, block2.Length);
+                Assert.Equal(expected, actual);
+
+                // Verify TransformFinalBlock
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                hash.TransformFinalBlock(block2, 0, block2.Length);
+                Assert.Equal(expected, hash.Hash);
+                Assert.Equal(expected, hash.Hash); // .Hash doesn't clear hash
+
+                // Todo: this will also fail due to the ActiveIssue below in VerifyUseAfterTransformFinalBlock
+                // but commented out so the other tests can run
+                // actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                // Assert.Equal(expected, actual);
+
+                // TransformBlock + TransformBlock + ComputeHash
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                hash.TransformBlock(block2, 0, block2.Length, null, 0);
+                actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        [Fact]
+        public void VerifyObjectDisposedException()
+        {
+            HashAlgorithm hash = Create();
+            hash.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => hash.Hash);
+            Assert.Throws<ObjectDisposedException>(() => hash.ComputeHash(Array.Empty<byte>()));
+            Assert.Throws<ObjectDisposedException>(() => hash.ComputeHash(Array.Empty<byte>(), 0, 0));
+            Assert.Throws<ObjectDisposedException>(() => hash.ComputeHash((Stream)null));
+            Assert.Throws<ObjectDisposedException>(() => hash.TransformBlock(Array.Empty<byte>(), 0, 0, null, 0));
+            Assert.Throws<ObjectDisposedException>(() => hash.TransformFinalBlock(Array.Empty<byte>(), 0, 0));
+        }
+
+        [Fact]
+        public void VerifyHashNotYetFinalized()
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                hash.TransformBlock(Array.Empty<byte>(), 0, 0, null, 0);
+                Assert.Throws<CryptographicUnexpectedOperationException>(() => hash.Hash);
+            }
+        }
+
+        [Fact]
+        [ActiveIssue(0)] // Todo: netfx treats HashFinal and Initialize separately, corefx doesn't and Initialize is a no-op
+        public void VerifyUseAfterTransformFinalBlock()
+        {
+            byte[] block = new byte[1] {1};
+
+            using (HashAlgorithm hash = Create())
+            {
+                hash.TransformBlock(block, 0, block.Length, null, 0);
+                hash.TransformFinalBlock(block, 0, block.Length);
+
+                // Todo: these do not throw on corefx, but do on netfx
```

Comment:
Doesn't seem like it would be hard to make this also throw on corefx; just requires tracking being in this state.